### ISSUE TITLE
`onkeypress` event panics when characters with `char_code` greater than `u8::MAX` are pressed

### ISSUE THREAD

**Issue by IsawU** [DioxusLabs/dioxus#621]
**Problem**

Whenever an `onkeypress` event fires with `KeyboardEvent.char_code` > `u8::MAX` the app panics. 

With `onkeydown` and `onkeyup` the characters are correctly returned in `KeyboardEvent.key` and `KeyboardEvent.char_code`.

The `€` Euro sign might be the most widely available character that causes a panic. All dead keys work as these never trigger an `onkeypress` event.

I use UTF-8 locale variant on my Linux.

<details>
<summary>Stack Traceback</summary>

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("invalid value: integer `8364`, expected u8", line: 0, column: 0)', /home/isawu/.cargo/registry/src/github.com-1ecc6299db9ec823/dioxus-desktop-0.2.3/src/events.rs:73:67
stack backtrace:
   0: rust_begin_unwind
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/panicking.rs:142:14
   2: core::result::unwrap_failed
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/result.rs:1785:5
   3: core::result::Result<T,E>::unwrap
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/result.rs:1107:23
   4: dioxus_desktop::events::make_synthetic_event
             at /home/isawu/.cargo/registry/src/github.com-1ecc6299db9ec823/dioxus-desktop-0.2.3/src/events.rs:73:23
   5: dioxus_desktop::events::trigger_from_serialized
             at /home/isawu/.cargo/registry/src/github.com-1ecc6299db9ec823/dioxus-desktop-0.2.3/src/events.rs:52:17
   6: dioxus_desktop::launch_with_props::{{closure}}::{{closure}}::{{closure}}
             at /home/isawu/.cargo/registry/src/github.com-1ecc6299db9ec823/dioxus-desktop-0.2.3/src/lib.rs:141:49
   7: core::option::Option<T>::map
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/option.rs:929:29
   8: dioxus_desktop::launch_with_props::{{closure}}::{{closure}}
             at /home/isawu/.cargo/registry/src/github.com-1ecc6299db9ec823/dioxus-desktop-0.2.3/src/lib.rs:138:25
   9: <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/alloc/src/boxed.rs:1954:9
  10: wry::webview::webkitgtk::InnerWebView::new::{{closure}}
             at /home/isawu/.cargo/registry/src/github.com-1ecc6299db9ec823/wry-0.13.3/src/webview/webkitgtk/mod.rs:86:11
  11: <O as webkit2gtk::auto::user_content_manager::UserContentManagerExt>::connect_script_message_received::script_message_received_trampoline
             at /home/isawu/.cargo/registry/src/github.com-1ecc6299db9ec823/webkit2gtk-0.17.1/src/auto/user_content_manager.rs:299:7
  12: g_closure_invoke
  13: <unknown>
  14: g_signal_emit_valist
  15: g_signal_emit
  16: <unknown>
  17: <unknown>
  18: <unknown>
  19: <unknown>
  20: <unknown>
  21: <unknown>
  22: <unknown>
  23: <unknown>
  24: <unknown>
  25: <unknown>
  26: <unknown>
  27: g_main_context_dispatch
  28: <unknown>
  29: g_main_context_iteration
  30: gtk_main_iteration_do
  31: gtk::auto::functions::main_iteration_do
             at /home/isawu/.cargo/registry/src/github.com-1ecc6299db9ec823/gtk-0.15.5/src/auto/functions.rs:405:24
  32: tao::platform_impl::platform::event_loop::EventLoop<T>::run_return::{{closure}}
             at /home/isawu/.cargo/registry/src/github.com-1ecc6299db9ec823/tao-0.6.4/src/platform_impl/linux/event_loop.rs:923:11
  33: glib::main_context::<impl glib::auto::main_context::MainContext>::with_thread_default
             at /home/isawu/.cargo/registry/src/github.com-1ecc6299db9ec823/glib-0.15.12/src/main_context.rs:156:12
  34: tao::platform_impl::platform::event_loop::EventLoop<T>::run_return
             at /home/isawu/.cargo/registry/src/github.com-1ecc6299db9ec823/tao-0.6.4/src/platform_impl/linux/event_loop.rs:817:5
  35: tao::platform_impl::platform::event_loop::EventLoop<T>::run
             at /home/isawu/.cargo/registry/src/github.com-1ecc6299db9ec823/tao-0.6.4/src/platform_impl/linux/event_loop.rs:773:21
  36: tao::event_loop::EventLoop<T>::run
             at /home/isawu/.cargo/registry/src/github.com-1ecc6299db9ec823/tao-0.6.4/src/event_loop.rs:177:5
  37: dioxus_desktop::launch_with_props
             at /home/isawu/.cargo/registry/src/github.com-1ecc6299db9ec823/dioxus-desktop-0.2.3/src/lib.rs:113:5
  38: dioxus_desktop::launch
             at /home/isawu/.cargo/registry/src/github.com-1ecc6299db9ec823/dioxus-desktop-0.2.3/src/lib.rs:48:5
  39: dioxus_test::main
             at ./src/main.rs:4:5
  40: core::ops::function::FnOnce::call_once
             at /rustc/897e37553bba8b42751c67658967889d11ecd120/library/core/src/ops/function.rs:248:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

</details>

**Steps To Reproduce**

- create an empty app
- with one input field with an `onkeypress` event
- try typing any diacritics or symbols available on your keyboard that are not available on the US layout

<details>
<summary>Minimal Example</summary>

```rust
use dioxus::{ prelude::* };

fn main() {
    dioxus::desktop::launch(app);
}

fn app(cx: Scope) -> Element {
    cx.render(rsx!(
        input {
            onkeypress: move |_e| {},
        },
    ))
}
```

```toml
[package]
name = "dioxus_test"
version = "0.1.0"
edition = "2021"

[dependencies]
dioxus = { version = "0.2.4", features = ["desktop"] }

```

</details>

**Expected behavior**

I'd expect for the program not to panic and correctly pass the characters to the callback.

**Environment:**
 - Dioxus version: `0.2.4`
 - Rust version: `1.65.0 (897e37553 2022-11-02)`
 - OS info: `Debian 11 (bullseye)`, `Microsoft Windows 10`
 - App platform: `desktop`

**Questionnaire**

- [x] I'm interested in fixing this myself but don't know where to start
- [ ] I would like to fix and I have a solution
- [ ] I don't have time to fix this right now, but maybe later

**Comment by IsawU** [comment]
`onkeydown` and `onkeyup` are not affected because they always have `"char_code": Number(0),` before parsing.

The problem with `onkeypress` originates from `KeyCode` being `#[repr(u8)]` and therefore `serde_repr` expects an `u8` value. Quickly changing `KeyCode` to be `#[repr(u32)]` fixes the issue, but it's most certainly not the correct fix.

**Comment by Demonthos** [comment]
This was fixed by #927
